src/lambda_handler.py

```diff
@@ -34,54 +164,196 @@ def task_default(event):
+        else:
+            file_header = [str(e) for e in file_header]
+    except Exception:
+        error_msg = "file_header must be a list of strings."
         logging.error(error_msg)
         failed_task(result, error_msg)
-    else:
+        return result
+
+    # Verify file_configs data, then initialize application regexes and a
+    # set to accumulate IPs for each file
+    for config in file_configs:
+        # Verify that required keys are present
+        for required_key in ["app_regex", "description", "filename"]:
+            if required_key not in config:
+                error_msg = 'Missing required key "%s" in file config: "%s"'
+                logging.error(error_msg, required_key, config)
+                failed_task(result, error_msg % (required_key, config))
+                return result
+
+        # Verify that app_regex is valid and initialize if it is
         try:
-            variable_error_msg = "numerator: %s, denominator: %s"
-            result["result"] = example_div(int(numerator), int(denominator))
-        except ValueError:
-            error_msg = "The provided values must be integers."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
-        except ZeroDivisionError:
-            error_msg = "The denominator cannot be zero."
-            logging.error(error_msg)
-            logging.error(variable_error_msg, numerator, denominator)
-            failed_task(result, error_msg)
+            config["app_regex"] = re.compile(config["app_regex"])
+        except (TypeError, re.error):
+            error_msg = 'Invalid app_regex "%s" provided in file config: "%s"'
+            logging.error(error_msg, config["app_regex"], config)
+            failed_task(result, error_msg % (config["app_regex"], config))
+            return result
+
+        # Initialize the set of static IPs
+        config["ip_set"] = {ip_network(i) for i in config.get("static_ips", [])}
+
+    # Name of the AWS resource tag whose value represents the application
+    # associated with an IP address
+    application_tag_name: str = event.get("application_tag", "Application")
+    # AWS resource tag name indicating whether an IP address should be published
+    publish_egress_tag_name: str = event.get("publish_egress_tag", "Publish Egress")
+    # Name of the IAM role to assume that can read the necessary EC2 data
+    # in each AWS account. Note that this role must exist in each account.
+    ec2_read_role_name: str = event.get("role_name", "EC2ReadOnly")
+
+    account_ids: List[str] = event.get("account_ids", None)
+    for account_id in account_ids:
+        # Verify AWS account ID is 12 digits
+        if not re.match(r"^\d{12}$", str(account_id)):
+            error_msg = 'Account ID "%s" is invalid - it must be 12 digits.'
+            logging.error(error_msg, account_id)
+            failed_task(result, error_msg % account_id)
```

Comment:
Should we return after this? Skip to the next account? I don't think we should just continue on if there's an issue with the account ID. A note if we skip is that any other account ID failures will overwrite the error message set by `failed_task()`.

Reply:
I'm fine with either. My only concern was continuing to work on an account that is invalid so whichever way you go will solve that 👍

Reply:
Remedied with https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/bba50bbc816877cc8434cef1873c9ac574ef03cd.

Reply:
Not to complicate things but thoughts on just making the account ID check a pre-flight thing like the bucket name? Optionally could build a list of invalid account IDs and `join()` them into the error message.

Reply:
Dang it, you're right - that is a superior solution.  I'll do that and let you know here when it's done.

Reply:
In https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/14c453181af74af74ccb6ba8e61b2a88e99ca92d, I check the account IDs prior to the main processing loop.  Arg, sorry @mcdonnnj - I forgot to give you co-commit credit, but thanks for the excellent suggestion.

As the number of pre-flight checks has increased, I decide to move all of them out of `task_publish()` and into a new function `validate_event_data()` with https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/f66553f29c61aaa7c0e7657ffab32a7f850e5ae8.

Reply:
@mcdonnnj - Can this conversation be resolved?